inc/blocks/ab-test/register.php

```diff
@@ -113,6 +113,30 @@ function render_block( array $attributes, ?string $inner_content = '' ) : string
 		$winner = $results['winner'] ?? false;
 	}
 
+	if ( is_preview() || isset( $_GET['set_test'] ) ) {
+		wp_enqueue_style(
+			'altis-experiments-features-blocks-abtest',
+			plugins_url( 'inc/blocks/ab-test/edit.css', Analytics\ROOT_FILE ),
+			[],
+			null
+		);
+		wp_enqueue_script(
+			'ab-test-preview',
+			Utils\get_asset_url( 'blocks/ab-test-preview.js' ),
+			[],
+			null
+		);
+
+		return sprintf(
+			'%s<div class="ab-test-xb-preview %s" data-client-id="%s" post-id="%s"><div class="ab-test-xb-preview__tabs" data-client-tabs="%3$s"></div><div class="ab-test-xb-preview__content" data-client-content="%3$s"></div></div>',
```

Comment:
The extra data attributes that repeat the client ID shouldn't be necessary, it can all be derived the scope you have in the `forEach()` loop over the blocks.

Additionally using `post-id` on it's own is invalid HTML, for custom attributes on regular HTML elements use data attributes e.g. `data-post-id`.

We use `post-id` as an attribute on the `<ab-test-block>` web component because we can define everything about a custom element, we can't define any new attributes for existing tags like `<div>`.

```suggestion
			'%s<div class="ab-test-xb-preview %s" data-client-id="%s" data-post-id="%s"><div class="ab-test-xb-preview__tabs"></div><div class="ab-test-xb-preview__content"></div></div>',
```

Reply:
The `post-id`/`data-post-id` is no longer needed with the changes made for the preview URL's so I have removed these now.